### ISSUE TITLE
Feature request: Support Saving to Multiple Destinations

### ISSUE THREAD

**Issue by KoriIku** [gildas-lormeau/SingleFile#1389]
**Is your feature request related to a problem? Please describe.**
I've noticed that SingleFile currently only allows for selecting a single save location, which can be limiting. This setup makes it difficult to save web content to multiple destinations at once, such as for long-term archiving, local browsing, and sharing with friends.

**Describe the solution you'd like**
I would love to see SingleFile support the ability to save to multiple destinations simultaneously. For example, it would be fantastic if I could save a webpage to my webdav for long-term storage, save a local copy for browsing, and also copy it to the clipboard for sharing with friends. This feature would greatly enhance the flexibility and convenience of using SingleFile.

**Describe alternatives you've considered (optional)**
One workaround is to manually save the webpage multiple times, each time selecting a different save location. However, this process is cumbersome and time-consuming, especially when dealing with multiple webpages.

**Additional context (optional)**
While SingleFile already supports all the necessary options for saving to different destinations, being able to select multiple destinations at once would be a fantastic addition. This enhancement would greatly improve the user experience and make SingleFile even more versatile. Thank you for considering this feature request!

**Comment by gildas-lormeau** [comment]
Thank you for the suggestion!

I agree that it would be useful. This feature requires some refactoring but it should be possible to implement it from a technical point of view.

**Comment by filviu** [comment]
Yes, this last question is my use case for this feature as well. I push most of the stuff to Karakeep but sometimes I just need a local snapshot.

**Comment by huyz** [comment]
@ak-42 @filviu You can already set up multiple profiles with different destinations and then right-click to the menu and select the profile you want. You can even toggle the default profile.

**Comment by huyz** [comment]
As for saving to multiple destinations at once, for my use case, it would be sufficient to either (1) have a "group profil"e that contains a list of other profiles to invoke or (2) support a new setting in profiles that lets the user specify the next profile to invoke (thus, a chain could be created).
Of course, the former is more flexible as it promotes reuse, but I could live with the latter since that seems fairly straightforward to implement.

**Comment by dionorgua** [comment]
@huyz Wow. Thanks for this. It's not that obvious because that popup menu item is hidden if only one profile is present. So first thing is to go to addon preferences and define second profile

**Comment by huyz** [comment]
Another tip: for those on a Mac who have the Keysmith app (or the Setapp subscription which includes it), you can set up a distinct keyboard shortcut for each profile.
Example:

![Image](https://github.com/user-attachments/assets/6c64bd18-4978-46ad-8855-e11e62c92689)
